var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -91,84 +110,55 @@ class Qt(Package):
-    depends_on("icu4c")
-    depends_on("fontconfig", when=(sys.platform != 'darwin'))  # (Unix only)
-    depends_on("freetype")
-    depends_on("sqlite", type=('build', 'run'))
     depends_on("pcre+multibyte", when='@5.0:5.8')
-    depends_on("pcre2+multibyte", when='@5.9:')
-    depends_on("double-conversion", when='@5.7:')
-    depends_on("harfbuzz", when='@5:')
-
-    # Core options:
-    # -doubleconversion  [system/qt/no]
-    # -iconv             [posix/sun/gnu/no] (Unix only)
-    # -pcre              [system/qt]
-
-    # Gui, printing, widget options:
-    # -harfbuzz          [system/qt/no]
-    # -xkbcommon-x11     [system/qt/no]
-    # -system-xkbcommon
-
-    # Database options:
-    # -sqlite            [system/qt]
-
-    # Qt3D options:
-    # -assimp            [system/qt/no]
-
-    # QtQml
-    depends_on("python", when='@5.7.0:', type='build')
-
-    # OpenGL hardware acceleration
-    depends_on("gl@3.2:", when='@4:+opengl')
-    # xcb is Linux-specific
-    depends_on("libxcb", when=sys.platform != 'darwin')
-    depends_on("xcb-util-image", when=sys.platform != 'darwin')
-    depends_on("xcb-util-keysyms", when=sys.platform != 'darwin')
-    depends_on("xcb-util-wm", when=sys.platform != 'darwin')
-    depends_on("xcb-util-renderutil", when=sys.platform != 'darwin')
-    depends_on("libxkbcommon", when=sys.platform != 'darwin')
     depends_on("inputproto", when='@:5.8')
-    depends_on("libx11", when=sys.platform != 'darwin')
+    depends_on("openssl@:1.0", when='@:5.9+ssl')
 
-    if sys.platform != 'darwin':
-        depends_on("libxext", when='@3:4.99')
+    depends_on("glib", when='@4:')
+    depends_on("libpng", when='@4:')
+    depends_on("dbus", when='@4:+dbus')
+    depends_on("gl@3.2:", when='@4:+opengl')
 
-    # Webkit
-    depends_on("flex", when='+webkit', type='build')
-    depends_on("bison", when='+webkit', type='build')
-    depends_on("gperf", when='+webkit')
+    depends_on("harfbuzz", when='@5:')
+    depends_on("double-conversion", when='@5.7:')
+    depends_on("pcre2+multibyte", when='@5.9:')
 
-    # Multimedia
-    # depends_on("gstreamer", when='+multimedia')
-    # depends_on("pulse", when='+multimedia')
-    # depends_on("flac", when='+multimedia')
-    # depends_on("ogg", when='+multimedia')
-    # -pulseaudio                [auto] (Unix only)
-    # -alsa                      [auto] (Unix only)
-
-    # Webengine options:
-    # -webengine-alsa            [auto] (Linux only)
-    # -webengine-pulseaudio      [auto] (Linux only)
-    # -webengine-embedded-build  [auto] (Linux only)
-    # -webengine-icu             [system/qt] (Linux only)
-    # -webengine-ffmpeg          [system/qt] (Linux only)
-    # -webengine-opus            [system/qt] (Linux only)
-    # -webengine-webp            [system/qt] (Linux only)
+    # Non-macOS dependencies and special macOS constraints
+    if MACOS_VERSION is None:
+        depends_on("fontconfig")
+        depends_on("libx11")
+        depends_on("libxcb")
```

Comment:
The original logic

```
depends_on("libxcb", when=sys.platform != 'darwin')
```

is a bit more idiomatic for Spack. This isn't particularly problematic though given that I assume people wouldn't be cross compiling this.

Reply:
I thought for `depends_on`,  `when` had to be a spec rather than a bool? Also, how would
```python
    if MACOS_VERSION is None:
        depends_on("libxext", when='@3:4.99')
```
be represented?

Reply:
Good point, I was wrong about that. Also:

* I incorrectly stated in https://github.com/spack/spack/pull/12138#discussion_r308453777 that you cannot have `when=True` - you can.
* However, in your case this would not work because it is mixing spec evaluation with booleans which is not supported

So in short this is fine.